datafusion/physical-optimizer/src/coalesce_batches.rs

```diff
@@ -92,3 +92,73 @@ impl PhysicalOptimizerRule for CoalesceBatches {
         true
     }
 }
+
+/// Remove CoalesceBatchesExec that are in front of a AggregateExec
+#[derive(Default, Debug)]
+pub struct UnCoalesceBatches {}
```

Comment:
I am wondering if we instead can avoid adding them in the `CoalesceBatches` optimizer

Reply:
Certainly something to attempt. I've not done it (yet) because it's not necessary to evaluate the impact of this change